lib/kubeclient/common.rb

```diff
@@ -506,16 +511,15 @@ def proxy_url(kind, name, port, namespace = '')
           @entities[kind.to_s].resource_name
         end
       ns_prefix = build_namespace_prefix(namespace)
-      rest_client["#{ns_prefix}#{entity_name_plural}/#{name}:#{port}/proxy"].url
+      "#{@api_endpoint}/#{@api_version}/#{ns_prefix}#{entity_name_plural}/#{name}:#{port}/proxy"
     end
 
     def process_template(template)
       ns_prefix = build_namespace_prefix(template[:metadata][:namespace])
       response = handle_exception do
-        rest_client["#{ns_prefix}processedtemplates"]
-          .post(template.to_h.to_json, { 'Content-Type' => 'application/json' }.merge(@headers))
+        http_client.post("#{ns_prefix}processedtemplates", template.to_h.to_json, json_headers)
       end
-      JSON.parse(response)
+      JSON.parse(response.body)
```

Comment:
The `rest_client`s `response` object returns its body when `to_s`'ed. With `faraday`'s `response`, need to call `.body` explicitely.